packages/relay/src/lib/eth.ts

```diff
@@ -1519,7 +1519,7 @@ export class EthImpl implements Eth {
     const maxGasLimit = constants.BLOCK_GAS_LIMIT;
     const gasUsed = blockResponse.gas_used;
 
-    if (contractResults?.results == null) {
+    if (contractResults == null) {
```

Comment:
When is this ever true with the use of pagination?
Since the result is an array at worst it'll be empty. So you should also check for emptiness

Reply:
good eye. the results response from pagination defaults to `[]`. gonna change this to check array length

Reply:
looking at this further and an empty array for contractResults is totally okay for blocks that don't have ContractCall, ContractCreate, or EthereumTransaction transactions

Reply:
Correct, however, you're utilizing the pagination logic which starts with `[]` and concats into it any results found.
So if no blocks are found the `const contractResults = await this.mirrorNodeClient.getContractResults()` will now return an empty array not a null response

Reply:
@bmino I still think this is an issue. A simple addition should do the trick
```suggestion
    if (contractResults == null || contractResults.length === 0) {
```

Reply:
I think I see the issue here. Logically this is a fail safe. A found may have no contract results, but we'd still want to send back the block response with no transactions in it, in which case we should return null on an empty set of contract results.
Will circle back to the null check but that's out of scope for your PR